go/proxyd/rpc.go

```diff
@@ -34,6 +35,17 @@ func (r *RPCErr) Error() string {
 	return r.Message
 }
 
+func IsValidID(id json.RawMessage) bool {
+	// handle the case where the ID is a string
+	if strings.HasPrefix(string(id), "\"") && strings.HasSuffix(string(id), "\"") {
```

Comment:
Why do we need this check if a string will be allowed by the next check?

Reply:
`IsValidID` is called with a `json.RawMessage`, which will be one of the following:

- A string value (with wrapping quotes)
- A number
- `true`, `false`, `null`
- An object or array.

I'm using the first check to make sure that if a string is passed in, it is a nonzero length. The subsequent check asserts  that if a string wasn't passed in, the ID is not an object or array. This matches the validation in Geth.